katello.gemspec

```diff
@@ -67,13 +67,13 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency "uglifier"
 
   # Testing
-  gem.add_development_dependency "factory_bot_rails", "~> 4.5"
-  gem.add_development_dependency "minitest-tags"
-  gem.add_development_dependency "mocha"
-  gem.add_development_dependency "vcr", "< 4.0.0"
-  gem.add_development_dependency "webmock"
-  gem.add_development_dependency "rubocop-checkstyle_formatter"
-  gem.add_development_dependency "simplecov"
-  gem.add_development_dependency "simplecov-rcov"
-  gem.add_development_dependency "robottelo_reporter"
+  #gem.add_dependency "factory_bot_rails", "~> 4.5"
+  gem.add_dependency "minitest-tags"
+  gem.add_dependency "mocha"
+  gem.add_dependency "vcr", "< 4.0.0"
+  gem.add_dependency "webmock"
+  gem.add_dependency "rubocop-checkstyle_formatter"
+  gem.add_dependency "simplecov"
+  gem.add_dependency "simplecov-rcov"
```

Comment:
That's a terrible hack, and I wonder how to "correctly" fix this.

`add_development_dependency` adds things to the `development` group of Bundler, but we do not install this group as we only want to run tests. There is no `add_test_dependency`.

I see three possible solutions, and not quite sure which one I like most:
1. install `development` group (and pull in all those really dev-only deps of foreman, meh)
2. set `development_group` to something else (like `${plugin}_dev`) when including the plugin in our actions, this would install all plugin dev deps as the new group is not excluded
3. provide a `gemfile.d/test.rb` that clearly defines the test deps in the `test` group.

I think *3* is the cleanest, but will also require the most changes (to every plugin that needs to define test deps)

Reply:
2 is what Jenkins does today, btw:
https://github.com/theforeman/jenkins-jobs/blob/6dd3bfe1fa010ecbb4750d9e909246a148c410fb/theforeman.org/pipelines/lib/foreman.groovy#L7